### PR TITLE
fix examples/layers/widgets/gestures.dart

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: examples/layers/widgets/gestures.dart [hunk 1]
```diff
@@ -104,42 +124,11 @@ class _GestureDemoState extends State<GestureDemo> {
...
-      else if (_swatch == Colors.deepOrange)
-        _swatch = Colors.brown;
-      else if (_swatch == Colors.brown)
-        _swatch = Colors.grey;
-      else if (_swatch == Colors.grey)
-        _swatch = Colors.blueGrey;
+      if (_swatch == _swatches.last) {
+        _swatch = _swatches.first;
+      } else {
+        _swatch = _swatches[_swatches.indexOf(_swatch) + 1];
```
Comment (Hixie) [inline examples/layers/widgets/gestures.dart]: This is certainly more readable, but for the record it's also a lot less performant (it's O(N), whereas the old logic was O(1)).

File: examples/layers/widgets/gestures.dart [hunk 2]
```diff
@@ -153,22 +142,24 @@ class _GestureDemoState extends State<GestureDemo> {
   Widget build(BuildContext context) {
     return new Stack(
       children: <Widget>[
-        new GestureDetector(
```
Comment (Hixie) [inline examples/layers/widgets/gestures.dart]: The better fix here is probably just to set the `fit` argument on `Stack` to `StackFit.expand`.

**Comment by Hixie** [comment]
Test?

File: examples/layers/widgets/gestures.dart (hunk 1 above)
Comment (a14n) [inline examples/layers/widgets/gestures.dart]: I changed the implementation to be more efficient and keep the array to be less error prone than the elseif serie.

File: examples/layers/widgets/gestures.dart (hunk 2 above)
Comment (a14n) [inline examples/layers/widgets/gestures.dart]: Thanks for the tip.
Done.

**Comment by a14n** [comment]
I will try to add tests later.

**Comment by a14n** [comment]
PTAL

File: examples/layers/widgets/gestures.dart [hunk 3]
```diff
@@ -104,42 +125,15 @@ class _GestureDemoState extends State<GestureDemo> {
...
-      else if (_swatch == Colors.brown)
-        _swatch = Colors.grey;
-      else if (_swatch == Colors.grey)
-        _swatch = Colors.blueGrey;
+      if (_swatch == kSwatches.last) {
+        _swatch = kSwatches.first;
+      } else {
+        // take the next color in _swatches list
+        _swatch = kSwatches
+            .skipWhile((MaterialColor swatch) => swatch != _swatch)
```
Comment (Hixie) [inline examples/layers/widgets/gestures.dart]: this is still O(N). Why not just remember the index of the selected color, and increment it by one? That would be O(1).

Comment (a14n) [inline examples/layers/widgets/gestures.dart]: I'm not expert in complexity evaluation but I don't see any difference between the new version and the old one - every values are compared.

Index added.

File: examples/layers/widgets/gestures.dart [hunk 4]
```diff
@@ -104,42 +126,10 @@ class _GestureDemoState extends State<GestureDemo> {
...
-        _swatch = Colors.orange;
-      else if (_swatch == Colors.orange)
-        _swatch = Colors.deepOrange;
-      else if (_swatch == Colors.deepOrange)
-        _swatch = Colors.brown;
-      else if (_swatch == Colors.brown)
-        _swatch = Colors.grey;
-      else if (_swatch == Colors.grey)
-        _swatch = Colors.blueGrey;
+      if (++_swatchIndex == kSwatches.length) {
```
Comment (Hixie) [inline examples/layers/widgets/gestures.dart]: nit: for clarity, do `_swatchIndex += 1` before the statement (conditions that mutate values can lead to confusion... here it's fine, but having any leads people to using them elsewhere and before you know it, you're reviewing some crypto code where someone is doing subtle value modification in the middle of a conditional).
also, no braces around single-line if statement bodies.

File: examples/layers/widgets/gestures.dart (hunk 3 above)
Comment (Hixie) [inline examples/layers/widgets/gestures.dart]: You're totally right, the old code was O(N) too. And unreadable. And impossible to maintain. Your fix is better in every dimension. :-)

**Comment by Hixie** [comment]
LGTM

So sorry about nit-picking this patch to death.

**Pull request merged**